lib/hydra/works/services/determine_mime_type.rb

```diff
@@ -0,0 +1,24 @@
+module Hydra::Works
+  class DetermineMimeType
+    # Determines the mime type for a given file
+    # @param [IO, File, Rack::Multipart::UploadedFile, #read] file
+    # @param [String, NilClass] original_name of the file
+    # @return [String]
+    class << self
+      def call(file, original_name = nil)
+        return file.mime_type if file.respond_to?(:mime_type) && file.mime_type.present?
+        return file.content_type if file.respond_to?(:content_type) && file.content_type.present?
+        mime_type_from_name_or_path(file, original_name) || default_mime_type
+      end
+
+      def default_mime_type
+        'application/octet-stream'
+      end
+
+      def mime_type_from_name_or_path(file, original_name)
+        return Hydra::PCDM::GetMimeTypeForFile.call(original_name) if original_name.present?
```

Comment:
This looks like a new condition (not seeing this above). Is this a bug fix change in addition to the refactor?